File: shared/feelings.py
"""
shared/feelings.py
------------------
Single source of truth for the trading feelings enum.

Used by:
  - api/schemas_trade.py   (field validation)
  - api/schemas_session.py (field validation)
  - api/models.py          (reference for column values)
"""
from __future__ import annotations

FEELING_VALUES: list[str] = [
    "calm",
    "focused",
    "confident",
    "anxious",
    "impatient",
    "fearful",
    "greedy",
    "distracted",
    "revenge",
    "tired",
]

POSITIVE_FEELINGS: frozenset[str] = frozenset({"calm", "focused", "confident"})
NEGATIVE_FEELINGS: frozenset[str] = frozenset(
    {"anxious", "impatient", "fearful", "greedy", "distracted", "revenge", "tired"}
)

File: ui/lib/feelings.ts
export type TradingFeeling =
  | "calm"
  | "focused"
  | "confident"
  | "anxious"
  | "impatient"
  | "fearful"
  | "greedy"
  | "distracted"
  | "revenge"
  | "tired";

export interface FeelingOption {
  value: TradingFeeling;
  label: string;
  sentiment: "positive" | "negative";
}

export const FEELING_OPTIONS: FeelingOption[] = [
  { value: "calm",        label: "Calm",        sentiment: "positive" },
  { value: "focused",     label: "Focused",     sentiment: "positive" },
  { value: "confident",   label: "Confident",   sentiment: "positive" },
  { value: "anxious",     label: "Anxious",     sentiment: "negative" },
  { value: "impatient",   label: "Impatient",   sentiment: "negative" },
  { value: "fearful",     label: "Fearful",     sentiment: "negative" },
  { value: "greedy",      label: "Greedy",      sentiment: "negative" },
  { value: "distracted",  label: "Distracted",  sentiment: "negative" },
  { value: "revenge",     label: "Revenge",     sentiment: "negative" },
  { value: "tired",       label: "Tired",       sentiment: "negative" },
];

export const FEELING_VALUES: TradingFeeling[] = FEELING_OPTIONS.map((f) => f.value);
